File: Apps/FCM/ios/NotificationServiceExtension/NotificationService.m
#import <NotificationServiceExtension-Swift.h>
#import "NotificationService.h"

@interface NotificationService ()

@property(nonatomic, strong) void (^contentHandler)(UNNotificationContent* contentToDeliver);
@property(nonatomic, strong) UNMutableNotificationContent* bestAttemptContent;

@end

@implementation NotificationService

// Create object of class MyAppNotificationServicePushHandler
MyAppNotificationServicePushHandler* nsHandlerObj = nil;

// Initialize the object
+ (void)initialize {
  nsHandlerObj = [[MyAppNotificationServicePushHandler alloc] init];
}

- (void)didReceiveNotificationRequest:(UNNotificationRequest*)request withContentHandler:(void (^)(UNNotificationContent* _Nonnull))contentHandler {
  [nsHandlerObj didReceive:request withContentHandler:contentHandler];
}

- (void)serviceExtensionTimeWillExpire {
  [nsHandlerObj serviceExtensionTimeWillExpire];
}

@end

File: Apps/FCM/ios/NotificationServiceExtension/NotificationService.swift
import UserNotifications
import CioMessagingPushFCM
import CioTracking

class NotificationService: UNNotificationServiceExtension {

    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
      // Configure SDK options as needed
      CustomerIO.initialize(siteId: Env.siteId, apiKey: Env.apiKey, region: .US) { config in
        config.autoTrackPushEvents = true
        config.logLevel = .debug
      }
      
      MessagingPush.shared.didReceive(request, withContentHandler: contentHandler)
    }
    
    override func serviceExtensionTimeWillExpire() {
      MessagingPush.shared.serviceExtensionTimeWillExpire()
    }
}
